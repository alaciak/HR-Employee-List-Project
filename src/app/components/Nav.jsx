import React from 'react';
import UserTypeSelection from '../containers/UserTypeSelection.jsx';
import { LanguageSelection } from './LanguageSelection.jsx';
import { Provider } from 'react-redux';
import store from '../Store.jsx';
import { I18n } from 'react-i18next';
import { translate } from 'react-i18next';

export class Nav extends React.Component {

  render() {
    const { t } = this.props;
    return (
      <header className='navigation'>
        <nav className='container'>
          <div className='row'>
            <div className='navigation-logo col-4'>HR <span>System </span>Management</div>
              <I18n ns="translations">
                {
                  (t, { i18n }) => (
                    <LanguageSelection i18n={ i18n }/>
                  )
                }
              </I18n>
              <div className='navigation-user col-6'>
                <p>{t('loggedUser')}:</p>
                <Provider store={ store }>
                  <UserTypeSelection />
                </Provider>
            </div>
          </div>
        </nav>
      </header>
    );
  }
}

export default (translate('translations')(Nav));
