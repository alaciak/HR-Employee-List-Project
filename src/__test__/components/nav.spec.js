import React from 'react';
import { Nav } from '../../app/components/Nav.jsx';
import { shallow, render } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Nav', () => {

//ignored to make them pass after i18n support
  xit('should render header and nav of the page', () => {

    const wrapper = shallow(<Nav />);

    expect(toJson(wrapper)).toMatchSnapshot();

  });

});
