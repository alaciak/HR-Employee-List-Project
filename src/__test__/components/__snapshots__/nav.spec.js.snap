// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render header and nav of the page 1`] = `
<header
  className="navigation"
>
  <nav
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="navigation-logo col-4"
      >
        HR 
        <span>
          System 
        </span>
        Management
      </div>
      <Translate(LanguageSelection) />
      <div
        className="navigation-user col-6"
      >
        <p>
          loggedUser
          :
        </p>
        <Provider
          store={
            Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            }
          }
        >
          <Connect(UserTypeSelection) />
        </Provider>
      </div>
    </div>
  </nav>
</header>
`;
